File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/widgetassertions/TreeViews_ViewModel.cs
/// <filename>
///     TreeViews_ViewModel.cs
/// </filename>
using System.Collections.Generic;
using widgetassertions;

namespace widgetassertions
{
    public class TreeViews_ViewModel
  {
        private int MyTreeViewElementsTreeSelectedRow;
     
        public     virtual int getMyTreeViewElementsTreeSelectedRow()
    {
      return this.MyTreeViewElementsTreeSelectedRow;
    }
     
        private bool isMyTreeViewElementsTreeVisible;
     
        public     virtual bool getIsMyTreeViewElementsTreeVisible()
    {
      return this.isMyTreeViewElementsTreeVisible;
    }
     
        private bool isMyTreeViewElementsTreeEnabled;
     
        public     virtual bool getIsMyTreeViewElementsTreeEnabled()
    {
      return this.isMyTreeViewElementsTreeEnabled;
    }
     
        private System.Collections.Generic.List<widgetassertions.TreeViews_ViewModelMyTreeViewElementsRow> MyTreeViewElementsTreeRows = new System.Collections.Generic.List<widgetassertions.TreeViews_ViewModelMyTreeViewElementsRow>();
     
        public     virtual System.Collections.Generic.List<widgetassertions.TreeViews_ViewModelMyTreeViewElementsRow> getMyTreeViewElementsTreeRows()
    {
      return this.MyTreeViewElementsTreeRows;
    }
     
        private bool isMyTreeViewElementsTreeDescriptionColumnVisible;
     
        public     virtual bool getIsMyTreeViewElementsTreeDescriptionColumnVisible()
    {
      return this.isMyTreeViewElementsTreeDescriptionColumnVisible;
    }
     
        private System.Collections.Generic.List<widgetassertions.TreeViews_ViewModelMyStringRowHandlesRow> MyStringRowHandlesTreeRows = new System.Collections.Generic.List<widgetassertions.TreeViews_ViewModelMyStringRowHandlesRow>();
     
        public     virtual System.Collections.Generic.List<widgetassertions.TreeViews_ViewModelMyStringRowHandlesRow> getMyStringRowHandlesTreeRows()
    {
      return this.MyStringRowHandlesTreeRows;
    }
     
        private string MyStringRowHandlesTreeSelectedRow;
     
        public     virtual string getMyStringRowHandlesTreeSelectedRow()
    {
      return this.MyStringRowHandlesTreeSelectedRow;
    }
     
        private System.Collections.Generic.List<widgetassertions.TreeViews_ViewModelMyParentIndexRowHandlesRow> MyParentIndexRowHandlesTreeRows = new System.Collections.Generic.List<widgetassertions.TreeViews_ViewModelMyParentIndexRowHandlesRow>();
     
        public     virtual System.Collections.Generic.List<widgetassertions.TreeViews_ViewModelMyParentIndexRowHandlesRow> getMyParentIndexRowHandlesTreeRows()
    {
      return this.MyParentIndexRowHandlesTreeRows;
    }
     
        private System.Collections.Generic.List<widgetassertions.TreeViews_ViewModelMyParentStringRowHandlesRow> MyParentStringRowHandlesTreeRows = new System.Collections.Generic.List<widgetassertions.TreeViews_ViewModelMyParentStringRowHandlesRow>();
     
        public     virtual System.Collections.Generic.List<widgetassertions.TreeViews_ViewModelMyParentStringRowHandlesRow> getMyParentStringRowHandlesTreeRows()
    {
      return this.MyParentStringRowHandlesTreeRows;
    }
  }
}

File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/widgetassertions/TreeViews_ViewModel.cpp

#include "TreeViews_ViewModel.h"
#include <vector>
#include <memory>
#include "TreeViews_ViewModelMyTreeViewElementsRow.h"
#include "TreeViews_ViewModelMyStringRowHandlesRow.h"
#include "TreeViews_ViewModelMyParentIndexRowHandlesRow.h"
#include "TreeViews_ViewModelMyParentStringRowHandlesRow.h"
#include <string>

namespace widgetassertions
{
  int TreeViews_ViewModel::getMyTreeViewElementsTreeSelectedRow( ) 
  {
    return this->MyTreeViewElementsTreeSelectedRow;
  }
  bool TreeViews_ViewModel::getIsMyTreeViewElementsTreeVisible( ) 
  {
    return this->isMyTreeViewElementsTreeVisible;
  }
  bool TreeViews_ViewModel::getIsMyTreeViewElementsTreeEnabled( ) 
  {
    return this->isMyTreeViewElementsTreeEnabled;
  }
  std::vector<std::shared_ptr<TreeViews_ViewModelMyTreeViewElementsRow>> TreeViews_ViewModel::getMyTreeViewElementsTreeRows( ) 
  {
    return this->MyTreeViewElementsTreeRows;
  }
  bool TreeViews_ViewModel::getIsMyTreeViewElementsTreeDescriptionColumnVisible( ) 
  {
    return this->isMyTreeViewElementsTreeDescriptionColumnVisible;
  }
  std::vector<std::shared_ptr<TreeViews_ViewModelMyStringRowHandlesRow>> TreeViews_ViewModel::getMyStringRowHandlesTreeRows( ) 
  {
    return this->MyStringRowHandlesTreeRows;
  }
  std::string TreeViews_ViewModel::getMyStringRowHandlesTreeSelectedRow( ) 
  {
    return this->MyStringRowHandlesTreeSelectedRow;
  }
  std::vector<std::shared_ptr<TreeViews_ViewModelMyParentIndexRowHandlesRow>> TreeViews_ViewModel::getMyParentIndexRowHandlesTreeRows( ) 
  {
    return this->MyParentIndexRowHandlesTreeRows;
  }
  std::vector<std::shared_ptr<TreeViews_ViewModelMyParentStringRowHandlesRow>> TreeViews_ViewModel::getMyParentStringRowHandlesTreeRows( ) 
  {
    return this->MyParentStringRowHandlesTreeRows;
  }
}

